SIPs/sip_001.md

```diff
@@ -0,0 +1,172 @@
+      sent to the Bitcoin network
+
+As this change needs to be implemented in version 0 of the API, its changes must
+not break the current endpoint's behavior. This exposes the following constraint:
+
+    - Using `num_editions` in `POST /api/pieces` and `POST /api/editions` still
+      needs to register a limited set of editions in a range of 0 to the value
+      of `num _editions`
+
+
+## Solution
+
+### Creation of unlimited editions
+
+Using `POST /api/pieces/1234/editions/` an API user can create a distinct set of unlimited
+editions using the following payload:
+
+```javascript
+POST /api/pieces/1234/editions/
+
+    {
+        "editions": [2, 1, 5, 59]
+    }
+```
+
+As mentioned in the requirements, here `editions` just specifies the
+edition numbers to be created. The numbers do not need to be ordered or
+consecutive.
+
+In response, the API returns a payload of the following form:
+
+
+```javascript
+202 Accepted
+
+    {
+        "notification": "We've succesfully received your request of registering
+        editions 1, 2, 5, 59 for piece 1234."
+        "websocket": "ws://ascribe.io/api/pieces/1234/editions"
+    }
+```
+
+In this case, a user of the API can use the `websocket` property of
+the response to listen in on the execution of the ongoing asynchronously
+executing task.
+Eventual mistakes of the API user's input, as for example submitting an edition
+number for an edition that has already been created in a previous request are
+all broadcasted only through this channel.
+Semantically wrong request content though, will only stop execution of the
+edition number in question, not the creation of all editions in the request.
+
+For example, assume the first request of this section went through successfully.
+Say, we make another request (btw. we do not need to wait for completion of the
+first request's tasks):
+
+
+```javascript
+POST /api/pieces/1234/editions/
+
+    {
+        "editions": [2]
+    }
+```
+
+Note, that this is the second time the API user requests the creation of
+edition 2 for piece 1234.
+Still, the API will respond with the above proposed schema (202 Accepted), but
+will eventually tell the API user via the provided websocket in the response that
+edition 2 could not have been registered, as it was already.
+
+
+### Freezing a set of unlimited editions
+
+Using `POST /api/pieces/1234/editions/`, submitting `num_editions` an API user
+is able to freeze/limit the unlimited editions of a piece.
+
+```javascript
+POST /api/pieces/1234/editions/
+
+    {
+        "num_editions": 59
+    }
+```
+
+Note that here `num_editions` needs to be at least greater than the greatest
+number used before when creating a set of unlimited editions.
+As we only registered the editions 1, 2, 5, 59 in previous example, this
+implies that when calling freeze on piece 1234, the server will create all
+remaining editions in the range of 2 - 58 (with the exception of 5).
```

Comment:
Again, is there a use case to create non sequential editions? This just seems to add complexity with no added value